extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/runtime/PanacheQueryImpl.java

```diff
@@ -37,6 +41,24 @@
 
     // Builder
 
+    @Override
+    public <T> PanacheQuery<T> project(Class<T> type) {
+        this.projections = new Document();
+        for (Method method : type.getDeclaredMethods()) {
+            if (method.getName().startsWith("get")) {
+                BsonProperty bsonProperty = method.getAnnotation(BsonProperty.class);
```

Comment:
I think the `@BsonProperty` can be on the (Java) field as well, not just the getter

Reply:
Ah, I see the description says this is not covered, disregard then

Reply:
@jmartisk I can easily support both, this is already done for the __main pojo__ so it's will be easily done for the _projected pojo__.
For the moment this is just a draft PR, if the design is accepted I will update it to also support `@BsonProperty` on fields.

Reply:
This is now supported on both fields and methods